crates/uv-trampoline/src/bounce.rs

```diff
@@ -33,27 +33,60 @@ use windows::Win32::{
 
 use crate::{eprintln, format};
 
-const MAGIC_NUMBER: [u8; 4] = [b'U', b'V', b'U', b'V'];
 const PATH_LEN_SIZE: usize = size_of::<u32>();
 const MAX_PATH_LEN: u32 = 32 * 1024;
 
-/// Transform `<command> <arguments>` to `python <command> <arguments>`.
+/// The kind of trampoline.
+enum TrampolineKind {
+    /// The trampoline should execute itself, it's a zipped Python script.
+    Script,
+    /// The trampoline should just execute Python, it's a proxy Python executable.
+    Python,
+}
+
+impl TrampolineKind {
+    const fn magic_number(&self) -> &'static [u8; 4] {
+        match self {
+            Self::Script => [b'U', b'V', b'S', b'C'],
+            Self::Python => [b'U', b'V', b'P', b'Y'],
+        }
+    }
+
+    fn from_buffer(buffer: &[u8]) -> Option<Self> {
+        if buffer.ends_with(Self::Script.magic_number()) {
```

Comment:
we need to add `UVUV` here to preserve compatibility with environments created pre-0.5.

Reply:
Why's that? Isn't the trampoline always self-contained?

Reply:
oh right, because we won't actually be cross reading newer entrypoints